test/test.js

```diff
@@ -2663,8 +2667,40 @@ describe('Page', function() {
       expect(responses[0].url()).toBe(server.EMPTY_PAGE);
       expect(responses[0].status()).toBe(200);
       expect(responses[0].ok()).toBe(true);
+      expect(responses[0].fromCache()).toBe(false);
+      expect(responses[0].fromServiceWorker()).toBe(false);
       expect(responses[0].request()).toBeTruthy();
     });
+
+    it('Response.fromCache()', async({page, server}) => {
+      const responses = new Map();
+      page.on('response', r => responses.set(r.url().split('/').pop(), r));
+
+      // Load and re-load to make sure it's cached.
+      await page.goto(server.PREFIX + '/cached/one-style.html');
+      await page.reload();
+
+      expect(responses.size).toBe(2);
+      expect(responses.get('one-style.html').status()).toBe(304);
+      expect(responses.get('one-style.html').fromCache()).toBe(false);
```

Comment:
Can we have a 200 code where it wasn’t cached?

Reply:
Do you mean instead of `304` we currently have?
This will make the test less isolated: if there's some other test that loads the same URL,
it will be cached.
For this reason the test explicitly reloads the page to trigger caching.

Reply:
In addition to. Have a css file that has cache busting.

Reply:
The cache-busting example is covered in the other place: see line 2670. Is it enough?

Reply:
Ah I missed that. All good